### ISSUE TITLE
Focus z-index [NDS-21]

### ISSUE THREAD

**Issue by sh0ji** [wwnorton/design-system#109]
Let's remove the default value for our focus indicator's z-index.

Our default focus indicator sets a `z-index` value to ensure that focus is visible when a focused element is behind another element (e.g., under a fixed header/footer). Without this, tabbing to an element that is behind another element would result in a failure of [2.4.7: Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html).

https://github.com/wwnorton/design-system/blob/16ddb6e5cc158f2594b44adcebfe062e3453305a/packages/core/src/a11y/index.scss#L52-L57

https://github.com/wwnorton/design-system/blob/16ddb6e5cc158f2594b44adcebfe062e3453305a/packages/core/src/elevation/tokens.scss#L18

For instance, if you <kbd>Tab</kbd> to a button on our website and then scroll, this will happen:
![image](https://user-images.githubusercontent.com/7939037/112855409-c297ba00-907c-11eb-9f4b-d01734c55123.png)

While this is a feature of our focus indicator that improves usability, we've gotten feedback from a few users that it is unexpected and I agree that we should rethink it.

[NDS-21]

[NDS-21]: https://wwnorton.atlassian.net/browse/NDS-21
